sklearnex/preview/covariance/covariance.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+# ===============================================================================
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+from scipy import sparse as sp
+from sklearn.covariance import EmpiricalCovariance as sklearn_EmpiricalCovariance
+from sklearn.utils import check_array
+
+from daal4py.sklearn._utils import sklearn_check_version
+from onedal.covariance import EmpiricalCovariance as onedal_EmpiricalCovariance
+
+from ..._device_offload import dispatch, wrap_output_data
+from ..._utils import PatchingConditionsChain
+
+
+class EmpiricalCovariance(sklearn_EmpiricalCovariance):
```

Comment:
Use n_jobs and hyperparameters decorators. For example, like in PCA: https://github.com/intel/scikit-learn-intelex/blob/5b308bec7ea7e65223d026c41370f1ae5ee05db1/sklearnex/preview/decomposition/pca.py#L46-L47